File: src/main/scala/akka/Application.scala
package akka

import Client.TestTransfer
import akka.actor.{ActorSystem, Props}
import akka.pattern.ask
import akka.util.Timeout

import scala.concurrent.duration._


object Application extends App {

  import system.dispatcher

  implicit val timeout = Timeout(5 seconds)

  val system = ActorSystem("BankAccountActorSystem")
  val client = system.actorOf(Props[Client], "client")
  val future = client ? TestTransfer
  future.map {
    result =>
      println(s"Result of TestTransfer: $result.")
      system.shutdown
  }

}

File: src/main/java/bankaccount/Application.java
package bankaccount;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;
import akka.pattern.Patterns;
import akka.util.Timeout;
import scala.concurrent.Await;
import scala.concurrent.duration.Duration;

public class Application {

    public static void main(String[] args) throws Exception {
        Timeout timeout = new Timeout(Duration.create(5, "seconds"));

        ActorSystem system = ActorSystem.create("BankAccountActorSystem");

        final ActorRef bankAccount = system.actorOf(Props.create(BankAccount.class), "bankAccount");
        System.out.println(Await.result(Patterns.ask(bankAccount, new Messages.Deposit(30000), timeout), timeout.duration()));
        System.out.println(Await.result(Patterns.ask(bankAccount, new Messages.Withdraw(20000), timeout), timeout.duration()));
        System.out.println(Await.result(Patterns.ask(bankAccount, new Messages.Withdraw(20000), timeout), timeout.duration()));

        system.shutdown();

    }

}
